### DISCUSSION TITLE
Newly created files have execute bit set

### DISCUSSION THREAD

**Discussion by kulak** [filebrowser/filebrowser#2548] [Q&A] [answered]
I don't quite understand why file creation mask is 755 and how I can enforce no execute bit on text files or on any files created from within filebrowser.

**Comment by JagerSprinkles** [comment]
I have the opposite issue. My files are created with 644 but I would like 755. Is there a config option for this? The documentation is not clear about this.

**Comment by kulak** [comment]
I don't remember anymore what I did, but most likely I had to control this through `systemd` configuration.

If your `systemd` service file is `filebrowser.service`, then as root edit service using:

    systemd edit filebrowser.service

and make sure to create the following section above the appropriate comment line.  The following is my configuration during edit:

```ini
### Editing /etc/systemd/system/filebrowser.service.d/override.conf
### Anything between here and the comment below will become the contents of the drop-in file

[Service]
ReadWritePaths=/var/lib/filebrowser /mnt/data/filebrowser
# 007 - restricts access to others
UMask=007

### Edits below this comment will be discarded
```

My configuration was deployed on `Arch` using AUR:

```
pacman -Qi filebrowser
Name            : filebrowser-bin
Version         : 2.25.0-1
URL             : https://github.com/filebrowser/filebrowser
Install Reason  : Explicitly installed
Install Script  : Yes
Validated By    : None
```

You can find service file shipped with the package on AUR website.

**Reply by kulak** [reply]
Here is a quick link to systemd service file on AUR  <https://aur.archlinux.org/cgit/aur.git/tree/filebrowser.service?h=filebrowser-bin>

**Comment by dncpax** [comment]
I am seeing the same issue: all files created are executable. Using docker so cant really change files... any pointers? Thanks.

**Reply by dncpax** [reply]
for a workaround I used the comand execution global settings to remove the execute bit:
After Copy:
chmod -x $DESTINATION

After Save:
chmod -x $FILE

After Upload:
chmod -x $FILE

This seems to be enough, even created files don't have the execute bit set.

**Reply by kulak** [reply]
systemd service unit's umask can control executable bit on created files

**Reply by thorinmesser** [reply]
On docker we would have to map /etc/systemd/system/filebrowser.service.d/ to some directory on the host. And I'm pretty sure edits would get lost when we update the image. A configuration to set this would be much better.

**Reply by kulak** [reply]
I missed the fact that context of discussion was Docker.  With docker things are not as easily controlled in this case.  Code fix provided  by Zouizoui78 shall address the issue.

**Comment by Zouizoui78** [comment, marked as answer]
Fixed by https://github.com/filebrowser/filebrowser/commit/5c5ab6b8750a5168f0ae2a26bd5de41e0b6d9637
